crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1045,6 +1056,12 @@ impl<'db> TypeInferenceBuilder<'db> {
         );
     }
 
+    fn set_return_type(&mut self, ty: Type<'db>, range: TextRange) {
```

Comment:
The name `set_return_type` suggests a single value we are setting, not pushing to a vector. I'd suggest `push_return_type` or `record_return_type`.